plutus-core/untyped-plutus-core/src/UntypedPlutusCore/Core/Instance/Flat.hs

```diff
@@ -205,22 +219,9 @@ instance ( Closed uni
          , Flat (Binder name)
          ) => Flat (Term name uni fun ann) where
     encode = encodeTerm
-    decode = decodeTerm NoLimit
+    decode = decodeTerm NoLimit (const True)
```

Comment:
Can we add a note here that this this is the default "unlimited"/"unchecked" flattening and is only used for local purposes? And that the "real deal" is the specialized decodeTerm/decodeProgram that are paremeterized by limit and predicate check

Reply:
Yeah. Maybe even we shouldn't have these instances, and we should instead only have them for specific newtype wrappers, like `Script`.